src/main/java/moze_intel/projecte/gameObjs/items/rings/SWRG.java

```diff
@@ -101,23 +102,21 @@ private void tick(ItemStack stack, EntityPlayer player)
 			}
 		}
 
-		float toRemove = 0;
+		if (player.getCapability(InternalTimers.CAPABILITY, null).canRingUpdate()) {
+			long toRemove = 0;
 
-		if (playerMP.capabilities.isFlying)
-		{
-			toRemove = 0.32F;
-		}
+			if (playerMP.capabilities.isFlying) {
+				toRemove = 6;
+			}
 
-		if (stack.getTagCompound().getInteger(TAG_MODE) == 2)
-		{
-			toRemove = 0.32F;
-		}
-		else if (stack.getTagCompound().getInteger(TAG_MODE) == 3)
-		{
-			toRemove = 0.64F;
-		}
+			if (stack.getTagCompound().getInteger(TAG_MODE) == 2) {
+				toRemove = 6;
+			} else if (stack.getTagCompound().getInteger(TAG_MODE) == 3) {
+				toRemove = 12;
```

Comment:
6 and 12 do not add up to 0.32 or 0.64

Reply:
Items are not final and need a better solution, this was only temp and done for the sake of testing